doc/source/serve/production-guide/best-practices.md

```diff
@@ -91,4 +99,26 @@ deployment_statuses:
   message: ''
 ```
 
-For Kubernetes deployments with KubeRay, tighter integrations of `serve status` with Kubernetes are available. See [Getting the status of Serve applications in Kubernetes](serve-getting-status-kubernetes).
\ No newline at end of file
+For Kubernetes deployments with KubeRay, tighter integrations of `serve status` with Kubernetes are available. See [Getting the status of Serve applications in Kubernetes](serve-getting-status-kubernetes).
+
+## Best practices for HTTP requests
+
+Most examples in these docs use straightforward `get` or `post` requests using Python's `requests` library, such as:
+
+```{literalinclude} ../doc_code/requests_best_practices.py
+:start-after: __prototype_code_start__
+:end-before: __prototype_code_end__
+:language: python
+```
+
+This pattern is useful for prototyping, but it isn't sufficient for production. In production, HTTP requests should use:
+
+* Retries: requests may occasionally fail due to transient issues (e.g. slow network, node failure, power outage, spike in traffic, etc.). Retry failed requests a handful (~10) of times to account for these issues.
```

Comment:
```suggestion
* Retries: Requests may occasionally fail due to transient issues (e.g., slow network, node failure, power outage, spike in traffic, etc.) Retry failed requests about 10 times to account for these issues.
```

Reply:
I'd rather remove 10 here, since I'm not sure if that's the ideal number across different use cases. Let me make the change.